contracts/contract_transactions.md

```diff
@@ -0,0 +1,68 @@
+[back](./contracts.md)
+## Contract Transactions
+
+Contract transactions are of three types:
+- Create
+- Call
+- Disable
+
+A contract is also a normal account with a balance.
+
+
+### Create Contract Transaction
+
+Anyone can create a new contract by submitting a create contract transaction.
+
+The transaction contains:
+- The address of the contract owner (the one signing and paying for the transaction) 
+- The byte code of the contract
+- The VM to use
+- A transaction fee
+- Optionally: A initial call to the contract including name, args and gas. (Can be used to emulate init function of Ethereum/Solidity contracts)
+
+
+
+```
+{ owner           :: public_key()
+, code            :: hex_bytes()
+, vm_version      :: hex_byte()
+, fee             :: amount()
+, init            :: none() | { gas :: amount()
```

Comment:
Instead of `none()` you mean `undefined`? I do not see how value can have type `none()`.

Reply:
Perhaps.
Well it is something like option(X) or perhaps use [] for optional?

We do not really have defined language for our transaction types or do we?

Suggestions are welcome.

Reply:
I really mean that the whole init field could be missing.

Reply:
> I really mean that the whole init field could be missing.

ah ok

Reply:
Don't use the none() type anyway, since it has a specific meaning. We can have a discussion whether it is good to have missing (optional) fields in a transaction at some point as well.

Reply:
Thinking a bit more about it it isn't clear whether it should be allowed to not run any code on contract creation.

Reply:
Ok so create will execute the contract and the create transaction will take a calldata field which for ring should make the contract run an init function.